Update SiFive U540 Platform to use new FdtLib

JasonBrave opened the pull request.

JasonBrave commented:
@JohnAZoidberg @vlsunil @leiflindholm @changab

leiflindholm commented:
There are other users of OpensbiPlatformLib in the tree that should be updated at the same time.

RiscVPlatformPkg.dsc can't be built at the moment due to something being outdated w.r.t. opensbi.
@vlsunil can you look into that bit?

And the Freedom U500 fails on a missing TimerLib mapping. @JohnAZoidberg ?

But could you @JasonBrave update the FdtLib mappings in those .dscs anyway as part of this? (For whatever reason there are two entries in the U500 one.)

JasonBrave commented:
> And the Freedom U500 fails on a missing TimerLib mapping
Fixed

> But could you @JasonBrave update the FdtLib mappings in those .dscs anyway as part of this? (For whatever reason there are two entries in the U500 one.)
Updated FdtLib mapping for both SiFive platforms

leiflindholm commented:
> Okay. Let me take a look. But I plan to remove this model of using opensbi as submodule completely.

That is an entirely acceptable solution :)
